docs/contributing/making-a-pull-request.md

```diff
@@ -0,0 +1,450 @@
+# Making a Pull Request
+
+You can contribute to Material for MkDocs by making a [pull request] that
+will be reviewed by maintainers and integrated into the main repository when
+the changes made are approved. You can contribute bug fixes, changes to the
+documentation, or new functionality you have developed.
+
+[pull request]: https://docs.github.com/en/pull-requests
+
+!!! note "Considering a pull request"
+
+    Before deciding to spend effort on making changes and creating a pull
+    request, please discuss what you intend to do. If you are responding to
+    what you think might be a bug, please issue a [bug report] first. If you
+    indend to work on documentation, create a [documentation issue]. If you
+    want to work on a new feature, please create a [change request].
+
+    Keep in mind the guidance given and let people advise you. It might be that
+    there are easier solutions to the problem you perceive and want to address.
+    It might be that what you want to achieve can already be done by
+    configuration.
+
+[bug report]: reporting-a-bug.md
+[documentation issue]: reporting-a-docs-issue.md
+[change request]: requesting-a-change.md
+
+## Learning about pull requests
+
+Before you consider making a pull request, you should familiarize yourself
+with the documentation on GitHub. The following articles are of particular
+importance:
+
+1. [Forking a repository]
+2. [Creating a pull request from a fork]
+3. [Creating a pull request]
+
+Note that they provide tailored documentation for different operating systems
+and different ways of interacting with GitHub. We do our best in the
+documentation here to describe the process as it applies to Material for MkDocs
+but cannot cover all possible combinations of tools and ways of doing things.
+It is also important that you understand the concept of a pull-request in
+general before continuing.
+
+[Forking a repository]: https://docs.github.com/en/get-started/quickstart/fork-a-repo
+[Creating a pull request from a fork]: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork
+[Creating a pull request]: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request
+
+!!! tip "Pay attention to the instructions"
```

Comment:
I have factored the first sentence into the introduction to the section. I do think it will help people to understand that when it comes to pull requests, it is the GitHub documentation they need to look at.

Confusingly, the Git CLI has a `--request-pull` command, which has nothing to do with a GitHub PR. That might be something worth pointing out?

Reply:
> Before you consider making a pull request, you should familiarize yourself
with the documentation on GitHub

> Note that they provide tailored documentation for different operating systems
and different ways of interacting with GitHub.

IMHO, it is clear enough that we're targeting GitHub. My guess is that pull requests are not that different on both platforms, any almost everybody should know GitHub + this guide gives them very detailed instructions further down the line. Thus I'd say the box is too generic and might rather confuse users.